docs/resolving.md

```diff
@@ -181,3 +181,70 @@ Resolved
   }
 }
 ```
+
+## Preserving root definitions
+When resolving a document it's possible to preserve root definitions like RAML 1.0 `types:`. This can be done by passing `true` boolean as the second argument to `.resolve()`. E.g. `.resolve(model, true)`.
```

Comment:
What other "root definitions" does it preserve aside from `types:`? I think we should list all of them here.

Reply:
It preserves traits, esourceTypes, annotationTypes and types. Updated the doc.